### PR TITLE
Search by transaction hash and account address, and show search suggestions

### PR DESCRIPTION
**Pull request by lukaw3d**
Preview https://0a69944a.lw-oasis-explorer-frontend.pages.dev/

Hover and focus states:
search

### CODE REVIEW

File: src/app/components/Search/search-utils.ts [hunk 1]
```diff
@@ -10,19 +11,18 @@ import { RouteUtils } from '../../utils/route-utils'
...
+    const blockHeight = validateAndNormalize.blockHeight(searchTerm)
+    const txHash = validateAndNormalize.txHash(searchTerm)
+    const evmAccount = validateAndNormalize.evmAccount(searchTerm)
+    const consensusAccount = validateAndNormalize.consensusAccount(searchTerm)
+    if (blockHeight) return RouteUtils.getBlockRoute(parseInt(blockHeight, 10), ParaTime.Emerald)
+    if (txHash) return RouteUtils.getTransactionRoute(txHash, ParaTime.Emerald)
+    if (evmAccount) return RouteUtils.getAccountRoute(evmAccount, ParaTime.Emerald)
+    if (consensusAccount) return RouteUtils.getAccountRoute(consensusAccount, ParaTime.Emerald)
+    // TODO: block hash, contract, validator, event
+    return undefined
```
Comment (csillag) [inline src/app/components/Search/search-utils.ts:25]: As far as I can tell, this is the code that determines the priority of the various things we can search for. Has this been defined somewhere?

**Comment by csillag** [comment]
Searching for an invalid term on the homepage doesn't result in any visible error messages. just warning on the console. Is that intentional?

**Comment by csillag** [comment]
Scratch that, false alarm, HRM artifact...

**Comment by csillag** [comment]
The placeholder in the search bar mentions "Tnx Hash, Transaction ID". I get the hash, but what is the Transaction Id, and can we really search for it?

**Review by csillag — APPROVED** [review]
Reviewed, LGTM.

File: src/app/components/Search/search-utils.ts (hunk 1 above)
Comment (lukaw3d) [reply src/app/components/Search/search-utils.ts:25]: Oh, no, this is just a simple implementation without supporting multiple results yet. It's not meant to prioritize search types, but just assume that they are all unique.

Multiple results will be a separate PR

**Pull request merged**